azext_edge/edge/commands_edge.py

```diff
@@ -224,6 +225,7 @@ def init(
         mq_broker_name=str(mq_broker_name),
         mq_authn_name=str(mq_authn_name),
         mq_insecure=mq_insecure,
+        dataflow_profile_instances=int(dataflow_profile_instances),
```

Comment:
why do we need to cast dataflow_profile_instances into int if the typing is already int?

Reply:
if you somehow get a string, it means you need to fix your param (arg_type=int or soemthing)

Reply:
The reason for this is when default values are assigned to the command handler, it isn't the native Python type but a `knack` equivalent. This can cause issues down the line in processing, if say you depend on isinstance() and you assume the native type. The code is continuously being refactored, but because of AIO's reliance on `init` at this point in time, I kept the same patterns as before (with other default args). 

If after reading, and it is still a concern, happy to discuss further.

> if you somehow get a string, it means you need to fix your param (arg_type=int or soemthing)

^-- the new parameter introduced in this pr has arg_type=int.